src/main/java/by/exadel/internship/service/InternshipService.java

```diff
@@ -16,24 +19,33 @@
 @RequiredArgsConstructor
 public class InternshipService {
 
+    private static final Logger logger =
+            LoggerFactory.getLogger("by.exadel.service");
+
     private final InternshipRepository internshipRepository;
     private final InternshipMapper internShipMapper;
 
     public GuestInternshipDTO getById(UUID uuid) {
-
+        MDC.clear();
+        MDC.put("className", InternshipService.class.getSimpleName());
+        logger.info("Try to get Internships with uuid=" + uuid);
         Internship internship = internshipRepository
                 .findById(uuid)
-                .orElseThrow(() -> new NotFoundException(String.format("Internship with id %s was not found!", uuid)));
-
+                .orElseThrow(() -> new NotFoundException("No such Internship with uuid = "+ uuid + " in DB"));
+        logger.info("Successfully got Internships with uuid=" + uuid);
         return internShipMapper.toGuestInternshipDTO(internship);
-
     }
 
     public List<GuestInternshipDTO> getAll() {
-
+        MDC.clear();
+        MDC.put("className", InternshipService.class.getSimpleName());
+        logger.info("Try to get all Internships");
         List<Internship> internships = internshipRepository.findAll();
+        if (internships.isEmpty()){
+            throw new NotFoundException("List of Internships is Empty");
```

Comment:
It's excessive. Notfound exception is only about 1 resource. It's normal situation when there is not found by some predicate. We just return empty list